integration-provisioning/src/test/java/org/symphonyoss/integration/provisioning/service/KeyPairServiceTest.java

```diff
@@ -0,0 +1,266 @@
+
+  @Mock
+  private IntegrationUtils utils;
+
+  @Spy
+  private IntegrationProperties properties = new IntegrationProperties();
+
+  @InjectMocks
+  private KeyPairService keyPairService = new KeyPairService(arguments);
+
+  @Before
+  public void init() throws IOException {
+    PowerMockito.mockStatic(Runtime.class);
+    PowerMockito.mockStatic(FileSystems.class);
+    PowerMockito.mockStatic(Files.class);
+
+    FileSystem fileSystem = mock(FileSystem.class);
+
+    when(FileSystems.getDefault()).thenReturn(fileSystem);
+    when(Files.getFileAttributeView(any(Path.class), eq(PosixFileAttributeView.class),
+        eq(LinkOption.NOFOLLOW_LINKS))).thenReturn(attributeView);
+    when(Runtime.getRuntime()).thenReturn(runtime);
+
+    doReturn(lookupService).when(fileSystem).getUserPrincipalLookupService();
+
+    List<String> optionValues = Arrays.asList(Boolean.TRUE.toString());
+    doReturn(optionValues).when(arguments).getOptionValues(GENERATE_CERTIFICATE);
+
+    doReturn(process).when(runtime).exec(anyString());
+    doReturn(process).when(runtime).exec(any(String[].class));
+
+    doReturn(new ByteArrayInputStream(new byte[]{})).when(process).getErrorStream();
+
+    Certificate certificateInfo = new Certificate();
+    certificateInfo.setCaKeyPassword(MOCK_KEY_PASSWORD);
+    certificateInfo.setCaKeyFile(MOCK_KEY_FILE);
+    certificateInfo.setCaCertFile(MOCK_CERT_FILE);
+    certificateInfo.setCaCertChainFile(MOCK_CERT_CA_CHAIN_FILE);
+
+    this.properties.setSigningCert(certificateInfo);
+
+    doReturn(StringUtils.EMPTY).when(utils).getCertsDirectory();
+  }
+
+  @Test
+  public void testShouldNotGenerateCertificate() throws InterruptedException {
+    doReturn(Collections.EMPTY_LIST).when(arguments).getOptionValues(GENERATE_CERTIFICATE);
+
+    keyPairService.exportCertificate(null, null);
+
+    verify(process, times(0)).waitFor();
+  }
+
+  @Test(expected = KeyPairException.class)
+  public void testFailGenerateKeyPair() throws IOException {
+    Application application = new Application();
+    application.setId(MOCK_APP_ID);
+    application.setComponent(MOCK_APP_TYPE);
+
+    doReturn(-1).when(process).exitValue();
+
+    keyPairService.exportCertificate(null, application);
+  }
+
+  @Test(expected = KeyPairException.class)
+  public void testFailGenerateCSR() throws IOException, InterruptedException {
+    Application application = new Application();
+    application.setId(MOCK_APP_ID);
+    application.setComponent(MOCK_APP_TYPE);
+
+    IntegrationSettings settings = new IntegrationSettings();
+    settings.setUsername(MOCK_APP_TYPE);
+
+    doReturn(0).when(process).exitValue();
+    doReturn(0).doThrow(IOException.class).when(process).waitFor();
+
+    keyPairService.exportCertificate(settings, application);
+  }
+
+  @Test(expected = KeyPairException.class)
+  public void testFailGenerateCertificate() throws IOException, InterruptedException {
+    Keystore keystore = new Keystore();
```

Comment:
Create common method for use in three locals, for example:

  private Keystore getKeystore() {
    Keystore keystore = new Keystore();
    keystore.setPassword(MOCK_KEY_PASSWORD);
    return keystore;
  }